extractor/image/image_test.go

```diff
@@ -102,6 +109,96 @@ func TestNewDockerImage(t *testing.T) {
 	}
 }
 
+func setupPrivateRegistry() *httptest.Server {
+	imagePaths := map[string]string{
+		"v2/library/alpine:3.10": "../testdata/alpine-310.tar.gz",
+	}
+	tr := registry.NewDockerRegistry(registry.Option{
+		Images: imagePaths,
+		Auth: auth.Auth{
+			User:     "test",
+			Password: "testpass",
+			Secret:   "secret",
+		},
+	})
+
+	return tr
+}
+
+func TestNewDockerImageWithPrivateRegistry(t *testing.T) {
+	tr := setupPrivateRegistry()
+	defer tr.Close()
+
+	serverAddr := tr.Listener.Addr().String()
+
+	token := jwt.NewWithClaims(jwt.SigningMethodHS256, jwt.MapClaims{
+		"iss": "testdocker",
+	})
+
+	registryToken, err := token.SignedString([]byte("secret"))
+	require.NoError(t, err)
+
+	type args struct {
+		imageName string
+		option    types.DockerOption
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    v1.Image
+		wantErr bool
```

Comment:
NIT: I'm usually asserting on the error message. This helps in making sure we have appropriate & meaningful error messages when troubleshooting

Reply:
Thank you for the feedback. I added the assert. I usually assert error messages, but I was slacked off a little 😄